src/main/scala/akka/io/AsyncDnsResolver.scala

```diff
@@ -43,36 +48,58 @@ class AsyncDnsResolver(cache: SimpleDnsCache, config: Config) extends Actor with
   private def nextId(): Short = {
     do {
       requestId = (requestId + 2).toShort
-    } while (requests.contains(requestId))
+    } while (requests.exists(_._1 == requestId))
```

Comment:
I've just trawled through some RFCs and I can't find any guidance on the use of IDs. I'm going to make a change such that we just increment a short value.